packages/components/src/components/notification/notification.ts

```diff
@@ -0,0 +1,344 @@
+
+      this.emit('sd-after-hide');
+    }
+  }
+
+  @watch('duration')
+  handleDurationChange() {
+    this.startAutoHide();
+  }
+
+  /** Shows the notification. */
+  async show() {
+    if (this.open) {
+      return undefined;
+    }
+
+    this.open = true;
+    return waitForEvent(this, 'sd-after-show');
+  }
+
+  /** Hides the notification */
+  async hide() {
+    if (!this.open) {
+      return undefined;
+    }
+
+    this.open = false;
+    return waitForEvent(this, 'sd-after-hide');
+  }
+
+  /**
+   * Displays the notification as a toast notification. This will move the notification out of its position in the DOM and, when
+   * dismissed, it will be removed from the DOM completely. By storing a reference to the notification, you can reuse it by
+   * calling this method again. The returned promise will resolve after the notification is hidden.
+   */
+  async toast() {
+    return new Promise<void>(resolve => {
+      const toastStack = this.toastStack === 'bottom-center' ? toastStackBottomCenter : toastStackDefault;
+
+      if (toastStack.parentElement === null) {
+        document.body.append(toastStack);
+      }
+
+      toastStack.appendChild(this);
+
+      // Wait for the toast stack to render
+      requestAnimationFrame(() => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-expressions -- force a reflow for the initial transition
+        this.clientWidth;
+        this.show();
+      });
+
+      this.addEventListener(
+        'sd-after-hide',
+        () => {
+          toastStack.removeChild(this);
+          resolve();
+
+          // Remove the toast stack from the DOM when there are no more alerts
+          if (toastStack.querySelector('sd-notification') === null) {
+            toastStack.remove();
+          }
+        },
+        { once: true }
+      );
+    });
+  }
+
+  render() {
+    return html`
+      <div
+        part="base"
+        class=${cx('w-full overflow-hidden flex items-stretch relative m-2')}
+        role="alert"
+        id="notification"
+        aria-hidden=${this.open ? 'false' : 'true'}
+        @mouseenter=${this.onHover}
+        @mouseleave=${this.onHoverEnd}
+      >
+        <slot
+          name="icon"
+          part="icon"
+          class=${cx(
+            'min-w-min flex items-center px-3 justify-center',
+            this.variant === 'info' && 'bg-info',
+            this.variant === 'success' && 'bg-success',
+            this.variant === 'warning' && 'bg-warning',
+            this.variant === 'error' && 'bg-error'
+          )}
+        >
+          <sd-icon
+            name=${this.variant === 'info'
+              ? 'info'
+              : this.variant === 'success'
+              ? 'success'
+              : this.variant === 'warning'
+              ? 'warning'
+              : this.variant === 'error'
+              ? 'error'
+              : ''}
+            library="system"
+            class="h-6 w-6 text-white"
+          ></sd-icon>
```

Comment:
Maybe it make sense to decouple the long if statemant?
Here's a cleaner and more readable version of your code:

```
const variantToName = {
  info: 'info',
  success: 'success',
  warning: 'warning',
  error: 'error',
};

const name = variantToName[this.variant] || '';

<sd-icon
  name=${name}
  library="system"
  class="h-6 w-6 text-white"
></sd-icon>
```